Source/turtlebot3/pubsub/ROS2ActionServerNode.cpp

```diff
@@ -0,0 +1,132 @@
+    FibonacciActionServer->ActionClass = UROS2FibonacciAction::StaticClass();
+
+    // bound callback function
+    FActionCallback UpdateFeedbackDelegate;
+    FActionCallback UpdateResultDelegate;
+    FActionGoalCallback HandleGoalDelegate;
+    FSimpleCallback HandleCancelDelegate;
+    FSimpleCallback HandleAcceptedDelegate;
+
+    UpdateFeedbackDelegate.BindDynamic(this, &AROS2ActionServerNode::UpdateFeedbackCallback);
+    UpdateResultDelegate.BindDynamic(this, &AROS2ActionServerNode::UpdateResultCallback);
+    HandleGoalDelegate.BindDynamic(this, &AROS2ActionServerNode::HandleGoalCallback);
+    HandleCancelDelegate.BindDynamic(this, &AROS2ActionServerNode::HandleCancelCallback);
+    HandleAcceptedDelegate.BindDynamic(this, &AROS2ActionServerNode::HandleAcceptedCallback);
+    FibonacciActionServer->SetDelegates(
+        UpdateFeedbackDelegate, UpdateResultDelegate, HandleGoalDelegate, HandleCancelDelegate, HandleAcceptedDelegate);
+
+    // // Add action server to ROS2Node
+    AddActionServer(FibonacciActionServer);
+}
+
+void AROS2ActionServerNode::UpdateFeedbackCallback(UROS2GenericAction* InAction)
+{
+    UROS2FibonacciAction* FibonacciAction = Cast<UROS2FibonacciAction>(InAction);
+
+    // Calculate fibonacci and send feedoback
+
+    // send result when finish by UpdateAndSendResult
+    if (Count++ < GoalRequest.order)
+    {
+        FeedbackMsg.sequence.Add(FeedbackMsg.sequence[Count] - FeedbackMsg.sequence[Count - 1]);
+        // Log request and response
+        UE_LOG(LogTurtlebot3,
+               Log,
+               TEXT("[%s][%s][C++][update feedback callback] added %d"),
+               *GetName(),
+               *ActionName,
+               FeedbackMsg.sequence[Count + 1]);
+    }
+    else
+    {
+        FibonacciActionServer->UpdateAndSendResult();
+    }
+}
+
+void AROS2ActionServerNode::UpdateResultCallback(UROS2GenericAction* InAction)
+{
+    UROS2FibonacciAction* FibonacciAction = Cast<UROS2FibonacciAction>(InAction);
+
+    // set result
+    FROSFibonacci_GetResult_Response ResultResponse;
+    ResultResponse.status = GOAL_STATE_SUCCEEDED;
+    ResultResponse.sequence = FeedbackMsg.sequence;
+    FibonacciAction->GetResultResponse(ResultResponse);
+
+    // stop timer
+    GetWorld()->GetTimerManager().ClearTimer(ActionTimerHandle);
+
+    // Log request and response
+    UE_LOG(LogTurtlebot3, Log, TEXT("[%s][%s][C++][update result callback]"), *GetName(), *ActionName);
+}
+
+bool AROS2ActionServerNode::HandleGoalCallback(UROS2GenericAction* InAction)
+{
+    UROS2FibonacciAction* FibonacciAction = Cast<UROS2FibonacciAction>(InAction);
+
+    // set and send goal response
+    auto* GoalResponse = (example_interfaces__action__Fibonacci_SendGoal_Response*)FibonacciAction->GetGoalResponse();
+    GoalResponse->accepted = true;
+    GoalResponse->stamp = UROS2Utils::FloatToROSStamp(UGameplayStatics::GetTimeSeconds(reinterpret_cast<UObject*>(GetWorld())));

```

Comment:
Could we use a `GetCurrentROS2Time()` defined [here](https://github.com/rapyuta-robotics/rclUE/blob/AddPublishTemplate/Source/rclUE/Public/rclcUtilities.h#L160), which is used also for sootball-ue.